app/gist/controller.js

```diff
@@ -68,16 +100,21 @@ export default Ember.Controller.extend({
   /**
    * Set the initial file columns
    */
-  initializeColumns: observer('model', function() {
-    var files = this.get('model.files');
+  initializeColumns: observer('model', 'numColumns', function() {
+    let files = this.get('model.files');
+    let numColumns = Math.min(this.get('numColumns'), MAX_COLUMNS);
 
-    if(files.objectAt(0)) {
-      this.set('col1File', files.objectAt(0));
+    let j = 0;
+    for (let i = 0; i < numColumns; ++i) {
```

Comment:
i would just make a single array that is a list of columns, and make the creation of that array be in a CP